src/models/IContract.js

```diff
@@ -334,6 +323,77 @@ class IContract {
       throw new Error('Contract is paused');
     }
   }
+
+  /**
+   * @function
+   * @description Load data from Web3Connection object,
+   * Called at start when testing or at login on MAINNET
+   */
+  _loadDataFromWeb3Connection() {
```

Comment:
I'm not understanding this function:

Why would any of these values change and then need to be assigned again? Why don't we simply assign these on the constructor and be done with it for life?

Reply:
A good example is "login" function, we are basically loading web3connection object, everything happens in it, so we usually need web3 object to connect to eth network and the account is we are using private key while testing.
***Mind that all those functions in Application.js like getStakingContract and getERC20Contract and similar, they are all used in a second step after instantiating Application and asserting it and then get our contracts objects using above functions, this issue was to solve exactly this, to remove that step, instantiate the contract we need without doing "app.getStakingContract()", this comes with a small price that the user does not need to know, we manage Web3Connection class to take care of everything else needed.